src/measure_theory/probability_mass_function.lean

```diff
@@ -303,4 +304,75 @@ by rw [← not_iff, filter_apply_eq_zero_iff, not_iff, not_not]
 def bernoulli (p : ℝ≥0) (h : p ≤ 1) : pmf bool :=
 of_fintype (λ b, cond b p (1 - p)) (nnreal.eq $ by simp [h])
 
+
+section outer_measure
+
+/-- Construct an `outer_measure` from a `pmf`, by assigning measure to each set `s : set α` equal
+  to the sum of `p x` for for each `x ∈ α` -/
+def to_outer_measure (p : pmf α) : measure_theory.outer_measure α :=
+measure_theory.outer_measure.sum
+  (λ (x : α), (p x) • (measure_theory.outer_measure.dirac x))
+
+@[simp]
+lemma to_outer_measure_apply (p : pmf α) (s : set α) :
+  p.to_outer_measure s = ∑' x, s.indicator (λ x, (p x : ℝ≥0∞)) x :=
+tsum_congr (λ x, (measure_theory.outer_measure.smul_dirac_apply (p x) x s))
+
+lemma to_outer_measure_apply' (p : pmf α) (s : set α) :
+  p.to_outer_measure s = ↑(∑' (x : α), s.indicator p x) :=
+by simp only [ennreal.coe_tsum (nnreal.indicator_summable (summable_coe p) s),
+  ennreal.coe_indicator, to_outer_measure_apply]
+
+@[simp]
+lemma to_outer_measure_apply_finset (p : pmf α) (s : finset α) :
+  p.to_outer_measure s = ∑ x in s, p x :=
+begin
+  refine (to_outer_measure_apply p s).trans ((@tsum_eq_sum _ _ _ _ _ _ s _).trans _),
+  { exact λ x hx, set.indicator_of_not_mem hx _ },
+  { exact finset.sum_congr rfl (λ x hx, set.indicator_of_mem hx _) }
+end
+
+lemma to_outer_measure_apply_eq_zero_iff (p : pmf α) (s : set α) :
+  p.to_outer_measure s = 0 ↔ disjoint p.support s :=
+begin
+  rw [to_outer_measure_apply', ennreal.coe_eq_zero,
+    tsum_eq_zero_iff (nnreal.indicator_summable (summable_coe p) s)],
+  exact function.funext_iff.symm.trans set.indicator_eq_zero',
+end
+
+@[simp]
+lemma to_outer_measure_caratheodory (p : pmf α) :
+  (to_outer_measure p).caratheodory = ⊤ :=
+begin
+  refine (eq_top_iff.2 $ le_trans (le_Inf $ λ x hx, _)
+    (measure_theory.outer_measure.le_sum_caratheodory _)),
+  obtain ⟨y, hy⟩ := hx,
+  exact ((le_of_eq (measure_theory.outer_measure.dirac_caratheodory _).symm).trans
+    (measure_theory.outer_measure.le_smul_caratheodory _ _)).trans (le_of_eq hy),
+end
+
+end outer_measure
+
+section measure
+
+variables [measurable_space α]
+
+/-- Since every set is Carathéodory-measurable under `pmf.to_outer_measure`,
+  we can further extend this `outer_measure` to a `measure` on `α` -/
+def to_measure (p : pmf α) : measure_theory.measure α :=
+p.to_outer_measure.to_measure ((to_outer_measure_caratheodory p).symm ▸ le_top)
+
+@[simp]
```

Comment:
This one shouldn't be a simp lemma: in general, when dealing with measures, most people don't want to see outer measures popping up. Instead, can you register lemmas saying that the measure of a set is the tsum of `p` of its elements, and the same thing for finsets? Also, measurability of the set should not be necessary here.

Reply:
I'm not sure how to avoid taking in the measurability of the set, unless there is someway to just assume that the `measurable_space α` instance is just `⊤`. Since we already have `p.to_outer_measure.caratheodory = ⊤` it seems like maybe just assuming the `measurable_space` instance makes everything measurable is reasonable, but I'm not sure how to do that in the code.

 It seems like this would require either implicit arguments, e.g. defining the type to be `pmf.to_measure p : @measure_theory.measure α ⊤`, or adding some kind additional mixin typeclass asserting that the `measurable_space α` instance is `⊤`, so I'm not sure how best to handle this.

Reply:
Yes, you are right, you really need the measurability of the set here. So forget my comment, sorry.